src/frontend/src/app/pages/proposal-review-edit/proposal-review-edit.component.ts

```diff
@@ -297,20 +352,38 @@ export class ProposalReviewEditComponent implements OnInit {
 
       this.commitForms.push(commitForm);
 
-      const subscription = commitForm.controls.reviewed.valueChanges.subscribe(
-        reviewed => {
+      const reviewedSubscription =
+        commitForm.controls.reviewed.valueChanges.subscribe(reviewed => {
           this.onCommitReviewedChange(reviewed, commitForm);
-        },
-      );
-      this.commitFormSubscriptions.push(subscription);
+        });
+      this.commitFormReviewedSubscriptions.push(reviewedSubscription);
+
+      const commitShaSubscription =
+        commitForm.controls.id.valueChanges.subscribe(commitSha => {
+          this.onCommitShaChange(commitSha, commitForm);
+        });
+      this.commitShaSubscriptions.push(commitShaSubscription);
+
+      this.focusCommitShaInput(this.commitForms.length - 1);
     }
   }
 
   public removeCommitForm(index: number): void {
     this.commitForms.splice(index, 1);
 
-    const [subscription] = this.commitFormSubscriptions.splice(index, 1);
-    subscription.unsubscribe();
+    const [reviewedSubscription] = this.commitFormReviewedSubscriptions.splice(
+      index,
+      1,
+    );
+    reviewedSubscription.unsubscribe();
+
+    const [commitShaSubscription] = this.commitShaSubscriptions.splice(
+      index,
+      1,
+    );
+    commitShaSubscription.unsubscribe();
+
+    this.focusCommitShaInput(Math.min(index, this.commitForms.length - 1));
```

Comment:
Focus the form group after the removed form group, or the last group if the last form group was removed